sotopia/database/evaluation_dimensions.py

```diff
@@ -0,0 +1,232 @@
+from redis_om import JsonModel
+from redis_om.model.model import Field
+from pydantic import BaseModel, create_model
+from typing import Type, Callable, Tuple, Annotated, Union
+
+
+class CustomEvaluationDimension(JsonModel):
+    name: str = Field(index=True)
+    description: str = Field(index=True)
+    range_high: int = Field(index=True)
+    range_low: int = Field(index=True)
+
+
+class CustomEvaluationDimensionList(JsonModel):
+    name: str = Field(index=True)
+    dimension_ids: list[str] = Field(default_factory=lambda: [], index=True)
+
+
+class EvaluationDimensionGenerator:
+    @staticmethod
+    def create_range_validator(low: int, high: int) -> Callable[[int], bool]:
+        def validator(x: Tuple[str, int]) -> Tuple[str, int]:
+            if not isinstance(x, tuple) or len(x) != 2:
+                raise ValueError("Must be a tuple of (str, int)")
+            if not isinstance(x[1], int) or not low <= x[1] <= high:
+                raise ValueError(f"Score must be between {low} and {high}")
+            return x
+
+        return validator
+
+    @staticmethod
+    def generate_dimension_model(dimension_ids: list[str]) -> Type[BaseModel]:
+        fields = {}
+
+        for dimension_id in dimension_ids:
+            dimension = CustomEvaluationDimension.get(dimension_id)
+            range_validator = EvaluationDimensionGenerator.create_range_validator(
+                dimension.range_low, dimension.range_high
+            )
+            field_type = Annotated[Tuple[str, int], range_validator]
+
+            fields[dimension.name] = (
+                field_type,
+                Field(..., description=dimension.description),
+            )
+
+        return create_model(
+            "CustomEvaluationDimensionModel",
+            __base__=BaseModel,
+            **fields,
+        )
+
+    @staticmethod
+    def generate_dimension_model_from_dict(
+        dimensions: list[dict[str, Union[str, int]]],
+    ) -> Type[BaseModel]:
+        fields = {}
+        for dimension_dict in dimensions:
+            dimension = CustomEvaluationDimension(**dimension_dict)
+            range_validator = EvaluationDimensionGenerator.create_range_validator(
+                dimension.range_low, dimension.range_high
+            )
+            field_type = Annotated[Tuple[str, int], range_validator]
+
+            fields[dimension.name] = (
+                field_type,
+                Field(..., description=dimension.description),
+            )
+
+        return create_model(
+            "CustomEvaluationDimensionModel",
+            __base__=BaseModel,
+            **fields,
+        )
+
+    @staticmethod
+    def generate_dimension_model_from_name(
```

Comment:
rename this? Also can we get rid of the printing here? And add the existing names to the doc?